pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -199,9 +199,25 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation, errorContext *er
 			Fn:           flow.TaskFn(botanist.DeployCloudProviderSecret).SkipIf(shootNamespaceInDeletion),
 			Dependencies: flow.NewTaskIDs(syncClusterResourceToSeed),
 		})
+		ensureShootStateExists = g.Add(flow.Task{
+			Name: "Ensuring that ShootState resource exists",
+			Fn:   flow.TaskFn(botanist.EnsureShootStateExists).RetryUntilTimeout(defaultInterval, defaultTimeout),
+		})
+		// In case ShootState was manually deleted we need to reload the persisted secrets from the Shoot's control plane
```

Comment:
Should change it to "in case the contents of ShootState get deleted" (although that's highly unlikely). At least that's why I left it after implementing the admission.